### ISSUE TITLE
mosaic and batch

### ISSUE THREAD

**Issue by alicera** [ultralytics/yolov5#3419]
## ❔Question
If I have 100 images to train. And the mosaic use 4th.
Batch size is 20, iteration is 5.
Every training image is compose of 4 images.
So the train images will be  25 (100/4), not 100 ?

**Comment by glenn-jocher** [comment]
@alicera yes mosaics are composed of 4 images by default.

**Comment by alicera** [comment]
Are there  25 images to train or 100 images?

**Comment by glenn-jocher** [comment]
@alicera yes all training images are used

**Comment by alicera** [comment]
1. If set the img-size is 640. Is the image size always 640 when network training process  ?
2. Is the images always 4th in one when network training process ?
   So the bellow status will not happen,right
 Assume the image size is 640x640
![_136450322413_20210602_174](https://user-images.githubusercontent.com/59388989/120793053-fbff1100-c568-11eb-8b76-ca5210207d97.jpg)
3. The padding color use (112,112,112) not (0,0,0,). Do you have reason?

**Comment by glenn-jocher** [comment]
@alicera training images are set to your --img-size, contents are augmented with scale jitter based on your hyperparameters.

Padding color is imagenet RGB mean.

**Comment by alicera** [comment]
Padding color is imagenet RGB mean.
Is It  better than black [0,0,0]?

**Comment by alicera** [comment]
How to process the bbox of woman face when mosaic?
It beyond the boundary.

original
![image](https://user-images.githubusercontent.com/59388989/121617645-00ab5400-ca98-11eb-90f3-93c3b8989668.png)
mosaic
![image](https://user-images.githubusercontent.com/59388989/121617685-13258d80-ca98-11eb-8a97-3d1a5b7ea3e5.png)

**Comment by alicera** [comment]
I find the np.clip(x, 0, 2 * s, out=x)  # clip when using random_perspective()
But it just set 0 and (height or width) .
How to filter the value when training？
